django-stubs/core/cache/backends/redis.pyi

```diff
@@ -0,0 +1,54 @@
+from collections.abc import Mapping
+from datetime import timedelta
+from typing import Any, Callable, Iterable, Protocol, SupportsInt, overload, type_check_only
+
+from _typeshed import ReadableBuffer, Unused
+from django.core.cache.backends.base import BaseCache
+from redis.client import Redis
+from redis.connection import BaseParser, ConnectionPool
+from typing_extensions import TypeAlias
+
+@type_check_only
+class RedisCacheClientSerializer(Protocol):
+    def dumps(self, obj: Any) -> bytes: ...
+    @overload
+    def loads(self, data: SupportsInt) -> int: ...
+    @overload
+    def loads(self, data: ReadableBuffer) -> Any: ...
```

Comment:
We might want to define a more simple signature

Reply:
What do you mean by "more simple"?

Reply:
I was just worried it might be painful to implement such a protocol if `RedisCacheClient.serializer` is explicitly specified, but this isn't that big of a concern

Reply:
I think most users would just opt to subclass the existing `RedisSerializer` concrete class.